contracts/DandelionOrg.sol

```diff
@@ -359,19 +423,30 @@ contract DandelionOrg is BaseTemplate {
         _createTokenManagerPermissions(_acl, tokenManager, dandelionVoting,  address(this));
     }
 
-    function _setupDandelionPermissions(ACL _acl) internal {
+    function _setupDandelionPermissions(
+        ACL _acl,
+        DandelionVoting dandelionVoting,
+        Redemptions redemptions,
+        TokenRequest tokenRequest,
+        TimeLock timeLock,
+        Delay delay,
+        TokenBalanceOracle tokenBalanceOracle,
+        DissentOracle dissentOracle
+    )
+        internal
+        {
 
         (TokenManager tokenManager, Vault agentOrVault) = _popBaseAppsCache();
-        (Voting dandelionVoting, Redemptions redemptions, TokenRequest tokenRequest, TimeLock timeLock, Delay delay, TokenBalanceOracle tokenBalanceOracle ) = _popDandelionAppsCache();
-
-        _createVotingPermissions(_acl, dandelionVoting, dandelionVoting, tokenManager, dandelionVoting);
-        _createRedemptionsPermissions(_acl, redemptions, tokenManager, agentOrVault, dandelionVoting, dandelionVoting);
-        _createTokenRequestPermissions(_acl, tokenRequest, tokenManager, dandelionVoting, dandelionVoting);
-        _createTimeLockPermissions(_acl, timeLock, dandelionVoting, tokenBalanceOracle, dandelionVoting);
-        _createDelayPermissions(_acl, delay, tokenManager, dandelionVoting, dandelionVoting);
-        _createTokenBalanceOraclePermissions(_acl, tokenBalanceOracle, dandelionVoting, dandelionVoting);
+
+        _createDandelionVotingPermissions(_acl, dandelionVoting, delay, timeLock, delay);
+        _createRedemptionsPermissions(_acl, redemptions, tokenManager, agentOrVault, delay, delay);
+        _createTokenRequestPermissions(_acl, tokenRequest, tokenManager, delay, delay);
+        _createTimeLockPermissions(_acl, timeLock, delay, tokenBalanceOracle, delay);
+        _createDelayPermissions(_acl, delay, dandelionVoting, delay);
+        _createTokenBalanceOraclePermissions(_acl, tokenBalanceOracle, delay, delay);
+        _createDissentOraclePermissions(_acl, dissentOracle, delay, delay);
         _createEvmScriptsRegistryPermissions(_acl, dandelionVoting, dandelionVoting);
-        _createVaultPermissions(_acl, agentOrVault, redemptions, dandelionVoting);
+        _createVaultPermissions(_acl, agentOrVault, redemptions, delay);
 
         _transferPermissionFromTemplate(_acl, tokenManager, dandelionVoting, tokenManager.MINT_ROLE(), dandelionVoting);
         _transferPermissionFromTemplate(_acl, tokenManager, dandelionVoting, tokenManager.BURN_ROLE(), dandelionVoting);
```

Comment:
Should be `redemptions` as grantee and `delay` as manager  instead of `dandelionVoting`